### PR TITLE
Add support for multiple simultaneous event scopes

### PR DESCRIPTION
**Pull request by tgoyne**
For HELP-40651. Requires https://github.com/realm/realm-core/pull/6288 so this'll currently fail to build on CI.

### CODE REVIEW

**Comment by tgoyne** [comment]
This can't be merged until we get a core release but it'd be good to get the code changes reviewed in advance so that we can ship a release asap.

**Review by leemaguire — COMMENTED** [review]
LGTM on first pass

**Review by jsflax — APPROVED** [review]

File: RealmSwift/Events.swift [hunk 1]
```diff
@@ -132,6 +179,15 @@ public extension Events {
             }
         }
     }
+}
+
+
```
Comment (dianaafanador3) [inline RealmSwift/Events.swift:184]: extra line

**Review by dianaafanador3 — APPROVED** [review]
LGTM! Just a small comment not that important

**Pull request merged**